docs/discovery-naming-conventions.md

```diff
@@ -0,0 +1,137 @@
+# Discovery Naming Conventions
+
+In order to load balance and route traffic to backend systems, Gimbal must
+discover the backends and sync them to the Gimbal cluster. This is done
+by the Gimbal discovery components, such as the Kubernetes discoverer and the
+OpenStack discoverer.
+
+During the discovery process, Gimbal translates the discovered backends into
+Kubernetes Services and Endpoints. The name of the discovered Services and
+Endpoints is called the _Discovered Name_, and is built from the following
+_Components_:
+
+```
+${discoverer-prefix}-${service-name}-${backend-name}
+```
+
+The name of service ports is not specified, and is handled independently by each
+discoverer implementation.
+
+## Kubernetes Service Naming Requirements
+
+Kubernetes Service names must adhere to the [rfc1035 DNS Label](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/identifiers.md) specification:
+
+> An alphanumeric (a-z, and 0-9) string, with a maximum length of 63 characters,
+> with the '-' character allowed anywhere except the first or last character,
+> suitable for use as a hostname or segment in a domain name.
+
+### Handling of invalid characters
+
+Backend systems other than Kubernetes (e.g. OpenStack) might support names that
+contain characters that are invalid according to the Kubernetes Service naming
+requirements. In these scenarios, discoverers will skip the service and inform
+the user via logs and metrics that the backend service name is not compatible
+with Gimbal.
+
+Even thought this might affect the user experience and cause friction, Gimbal
```

Comment:
Not sure if it applies to this proposal, but to document the places/ways to find these errors? E.g. metrics in Prometheus will alert --> then need to tail logs from Discoverers? This is where having the `Discovery Operator` we discussed can help. Having a CRD for each Discoverer gives us a place to dump errors to and you can look at the errors placeholder in that object easily find issues.